### PR TITLE
[CI] Enable Cuda/OpenCL e2e on PR comment

### PR DESCRIPTION
**Pull request by wlemkows**
- [x] review fixes
- [x] trigger Cuda e2e on an event (commnet, label or sth.)

### CODE REVIEW

File: .github/workflows/e2e_core.yml [hunk 1]
```diff
@@ -1,9 +1,26 @@
-name: E2E Nightly
+name: E2E build & run
 
 on:
-  schedule:
-    # Run every day at 23:00 UTC
-    - cron: '0 23 * * *'
+  workflow_call:
```
Comment (pbalcer) [inline .github/workflows/e2e_core.yml:4]: I'm afraid you will have to add some comments here to describe this incantation (and the one below) ;)

**Comment by lukaszstolarczuk** [comment]
@wlemkows, FYI, check out this PR: #1294

File: .github/workflows/e2e_core.yml [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 
 on:
-  schedule:
-    # Run every day at 23:00 UTC
-    - cron: '0 23 * * *'
+  # this workflow can by only triggered by other workflows
+  # for example by: e2e_cuda.yml or e2e_opencl.yml
+  workflow_call:
+    # acceptable input from adapter-specific workflows
+    inputs:
```
Comment (lukaszstolarczuk) [inline .github/workflows/e2e_core.yml:8]: pls add `description:` for each input

File: .github/workflows/e2e_core.yml (hunk 1 above)
Comment (wlemkows) [reply .github/workflows/e2e_core.yml:6]: Done.

File: .github/workflows/e2e_core.yml (hunk 2 above)
Comment (wlemkows) [reply .github/workflows/e2e_core.yml:8]: Done.

**Review by pbalcer — APPROVED** [review]

**Pull request merged**